lambdas/webhook.py

```diff
@@ -188,11 +185,30 @@ def handle_job_end_webhook(message_body: dict) -> dict:
     }
 
 
+def get_job_type(job_name: str) -> tuple[str, Callable]:
+    """Given an expected job name from the Alma webhook POST request, return the job type
+    and corresponding function for generating the step
+    function input.
+    """
+    if job_name.startswith(os.environ["ALMA_POD_EXPORT_JOB_NAME"]):
+        logger.info("PPOD export job webhook received.")
+        return "PPOD", generate_ppod_step_function_input
+    elif job_name.startswith(os.environ["ALMA_TIMDEX_EXPORT_JOB_NAME_PREFIX"]):
+        logger.info("TIMDEX export job webhook received.")
+        return "TIMDEX", generate_timdex_step_function_input
+    elif job_name.startswith(os.environ["ALMA_BURSAR_EXPORT_JOB_NAME_PREFIX"]):
+        logger.info("BURSAR export job webhook received.")
+        return "BURSAR", generate_bursar_step_function_input
+    else:
+        raise ValueError(job_name)
```

Comment:
While doing some testing, ran into the situation where the new image was deployed in `Dev1` ECR, but the env var `ALMA_BURSAR_EXPORT_JOB_NAME_PREFIX ` had not been set yet, which we chatted about on slack.  As-is, this prevents running some of the manual post-deploy tests in the README until the env var is added; which was on the brain, as we're discussing updating them as well.

At the risk of being overly prescriptive, what if we updated this function to allow missing env vars?  This feels like a pretty critical part of this lambda's functionality, and maybe this could also help decouple the values getting check and functions returned from the checking logic.

I was conflicted, as the handler throwing an error when the env var isn't set is also helpful.  But I think, as-is, that possibility is hidden in the event either of the first two checks -- POD or TIMDEX -- hit.

Maybe something like:

```python
def get_job_type(job_name: str) -> tuple[str, Callable]:
    """Given an expected job name from the Alma webhook POST request, return the job type
    and corresponding function for generating the step function input.
    """
    job_types = [
        (
            "PPOD",
            "ALMA_POD_EXPORT_JOB_NAME",
            generate_ppod_step_function_input,
            "PPOD export job webhook received.",
        ),
        (
            "TIMDEX",
            "ALMA_TIMDEX_EXPORT_JOB_NAME_PREFIX",
            generate_timdex_step_function_input,
            "TIMDEX export job webhook received.",
        ),
        (
            "BURSAR",
            "ALMA_BURSAR_EXPORT_JOB_NAME_PREFIX",
            generate_bursar_step_function_input,
            "BURSAR export job webhook received.",
        ),
    ]

    for job_type, env_key, step_function_input_handler, log_msg in job_types:
        job_name_prefix = os.getenv(env_key)
        if not env_key:
            logger.warning("expected env var not present: %s" % env_key)
        if job_name_prefix and job_name.startswith(job_name_prefix):
            logger.info(log_msg)
            return job_type, step_function_input_handler

    # if job type not matched, raise exception
    raise ValueError(job_name)
```

Reply:
I like this approach to how the function works. I'm not sure I understand the benefit of checking for the env vars here (and allowing for missing ones) over checking earlier like we do for the WORKSPACE env var, and raising a runtime error if anything is missing. https://github.com/MITLibraries/alma-webhook-lambdas/blob/35cf24e20ad128583b687d71682347f40f5632df/lambdas/webhook.py#L34

Also, I'm a little confused on whether we can test changes locally in dev1 without updating the terraform (if the changes would require it, e.g. a new env var is added). It seems like we should be able to do this...